components/script/dom/filereader.rs

```diff
@@ -491,7 +497,7 @@ fn perform_annotated_read_operation(
     let task = FileReadingTask::ProcessRead(filereader.clone(), gen_id);
     task_source.queue_with_canceller(task, &canceller).unwrap();
 
-    let task = FileReadingTask::ProcessReadData(filereader.clone(), gen_id);
+    let task = FileReadingTask::ProcessReadData(filereader.clone(), gen_id, blob_contents.clone());
```

Comment:
I suggest instead of passing the blob_contents to this task, we check whether blob_contents is empty and skip it entirely. This will make many of the other changes in this PR unnecessary.